src/viking.js

```diff
@@ -1,16 +1,63 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.damage -= 0;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health <= 0) {
+      return this.name + ' has died in act of combat';
+    } else {
+      return this.name + ' has received ' + damage + ' points of damage';
+    }
+  }
+  battleCry() {
+    return 'Odin Owns You All!';
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health <= 0) {
+      return 'A Saxon has died in combat';
+    } else {
+      return 'A Saxon has received ' + damage + ' points of damage';
+    }
+  }
+}
 
 // War
-class War {}
-
+class War {
+  vikingArmy = [];
+  saxonArmy = [];
```

Comment:
good start in the bonus! Please take a look in the solutions and if you have any questions, you can ask us!